hal-sim/hal_impl/fndef.py

```diff
@@ -206,20 +212,27 @@ def gen_func(f, name, restype, params, out, _thunk):
             """
         %s
         def %s(%s):
-            %s
-            return_value = %s(%s)
-            %s
-            return return_value
+            try:
+                %s
+                return_value = %s(%s)
+                %s
+            except NotImplementedError:
+                raise
+            except Exception as e:
+                raise SimulationError("unexpected exception calling '%s'") from e
```

Comment:
Do you think we should also pass through any HALErrors we get?

Reply:
I suppose, though they aren't used in simulation anywhere other than here.